contrib/Shopify/sarama/sarama.go

```diff
@@ -208,9 +208,11 @@ func WrapAsyncProducer(saramaConfig *sarama.Config, p sarama.AsyncProducer, opts
 				if headersSupported && saramaConfig.Producer.Return.Successes {
 					spans[tracer.SpanID(span.Context())] = span
 				} else {
-					// if returning successes isn't enabled, we just finish the
-					// span right away because there's no way to know when it will
-					// be done
+					// If returning successes isn't enabled, we just finish the span right away,
+					// because there's no way to know when it will be done.
+					// Note: Partition and Offset fields should not be accessed
```

Comment:
Sorry, I mean to add this to the entire function (WrapAsyncProducer) above.

Reply:
Not sure if it is worth adding to the doc comment . This is the property of the instrumented library itself. There is also  `Timestamp` field that shares the same "property"

Reply:
I think it is important we add it so contributors in future don't make the same mistake again.

Reply:
https://github.com/signalfx/signalfx-go-tracing/pull/128/commits/c92e3a5a13d833f82506a70eecffe48a475175b3